x/auth/vesting/msg_server.go

```diff
@@ -158,3 +160,57 @@ func (s msgServer) CreatePeriodicVestingAccount(goCtx context.Context, msg *type
 	return &types.MsgCreatePeriodicVestingAccountResponse{}, nil
 
 }
+
+func (s msgServer) DonateAllVestingTokens(goCtx context.Context, msg *types.MsgDonateAllVestingTokens) (*types.MsgDonateAllVestingTokensResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	ak := s.AccountKeeper
+	dk := s.DistrKeeper
+
+	from, err := sdk.AccAddressFromBech32(msg.FromAddress)
+	if err != nil {
+		return nil, err
+	}
+
+	acc := ak.GetAccount(ctx, from)
+	if acc == nil {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "account %s not exists", msg.FromAddress)
+	}
+
+	vestingAcc, ok := acc.(exported.VestingAccount)
+	if !ok {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "account %s not vesting account", msg.FromAddress)
+	}
+
+	if !vestingAcc.GetDelegatedVesting().IsZero() {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "account %s has delegated vesting tokens", msg.FromAddress)
+	}
+
+	vestingCoins := vestingAcc.GetVestingCoins(ctx.BlockTime())
+	if vestingCoins.IsZero() {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "account %s has no vesting tokens", msg.FromAddress)
+	}
+
+	// Change the account as normal account
+	ak.SetAccount(ctx,
```

Comment:
Is it okay to change the vesting account to  base account before FundCommunityPool?

Reply:
We can fund the community pool after we change the vesting account to the base account.
If not, we cannot send the fund due to vesting.

Reply:
I see, got it